File: internal/content/testdata/bench/go/modular-exponentiation.go
func modular_exponentiation(x, n, m int) int {
	res := 1
	x = x % m
	for n > 0 {
		if n%2 != 0 {
			res = (res * x) % m
			n--
		} else {
			x = (x * x) % m
			n /= 2
		}
	}
	return res
}

File: internal/content/testdata/bench/rust/modular-exponentiation.rs
fn modular_exponentiation(x: i64, n: i64, m: i64) -> i64 {
    let mut res: i64 = 1;
    let mut x = x % m;
    let mut n = n;
    while n > 0 {
        if n % 2 != 0 {
            res = (res * x) % m;
            n -= 1;
        } else {
            x = (x * x) % m;
            n /= 2;
        }
    }
    res
}
